### PR TITLE
add 12 missing NODE_FIELDS

### LINKED ISSUE #13558: [Bug]: variance is missing from NODE_FIELDS.ClassProperty
### 💻

- [X] Would you like to work on a fix?

### How are you using Babel?

Programmatic API (`babel.transform`, `babel.parse`)

### Input code

```js
console.log(require('@babel/types').NODE_FIELDS.ClassProperty.variance)
```

### Configuration file name

.babelrc

### Configuration

```json
{
  "presets": [["@babel/preset-env", { "targets": { "node": "current" } }], "@babel/preset-flow"]
}
```

### Current and expected behavior

Expected `NODE_FIELDS.ClassProperty.variance` to be defined, but it isn't.

Parsing the following code produces a `ClassProperty` with a `variance` field:

```js
class Foo {
  +blah: number 
}
```
https://astexplorer.net/#/gist/40f552d7977ffe512792d48ccb584ced/a9dcd37f91fc623481253f94835888b04c744136

### Environment

- `@babel/types` version: 7.14.5

### Possible solution

copy over the `variance` field definition from `ObjectTypeProperty` to `ClassProperty`

### Additional context

_No response_

### LINKED ISSUE #13563: [Bug]: readonly, declare, and definite are missing from NODE_FIELDS.ClassPrivateProperty
### 💻

- [X] Would you like to work on a fix?

### How are you using Babel?

Programmatic API (`babel.transform`, `babel.parse`)

### Input code

```js
console.log(require('@babel/types').NODE_FIELDS.ClassPrivateProperty)
```

### Configuration file name

_No response_

### Configuration

```json
{
  "presets": [["@babel/preset-env", { "targets": { "node": "current" } }], "@babel/preset-typescript"]
}
```

### Current and expected behavior

Expected `NODE_FIELDS.ClassPrivateProperty` to have `readonly`, `declare`, and `definite` fields, but it doesn't.

This parses, so I assume `ClassPrivateProperty` should have these fields:

```
class Foo {
	declare readonly #bar!
}
```

### Environment

`@babel/types`: 7.14.5

### Possible solution

Copy over all the missing fields from `ClassProperty`.

### Additional context

### PR DESCRIPTION
**Pull request by jedwards1211**
fix #13558
fix #13563

QUESTION: Can `ExportDefaultDeclaration.exportKind` be anything other than `"value"`?  I looked through the parser code and tried to come up with Flow or TS example that would result in `exportKind: "type"`, but couldn't find any.

| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | `Fixes #1, Fixes #2` 
| Patch: Bug Fix?          | Yes 
| Major: Breaking Change?  | No
| Minor: New Feature?      | No
| Tests Added + Pass?      | Yes
| Documentation PR Link    | 
| Any Dependency Changes?  | Added `glob` and `fs-extra` to `devDependencies` of `@babel/types`
| License                  | MIT

<a href="https://gitpod.io/#https://github.com/babel/babel/pull/13577"></a>

### CODE REVIEW

File: packages/babel-types/package.json [hunk 1]
```diff
@@ -30,7 +30,9 @@
   "devDependencies": {
     "@babel/generator": "workspace:*",
     "@babel/parser": "workspace:*",
-    "chalk": "^4.1.0"
+    "chalk": "^4.1.0",
+    "fs-extra": "^10.0.0",
```
Comment (JLHwung) [inline packages/babel-types/package.json:34]: I guess we can replace `fs-extra` by `JSON.parse(fs.readFileSync(#))`?

Comment (jedwards1211) [reply packages/babel-types/package.json]: I just switched it to `promisify(fs.readFile)`, wouldn't want to block the entire process

**Comment by jedwards1211** [comment]
There are lint errors I didn't cause:

```
Error: packages/babel-traverse/src/path/lib/virtual-types.ts(119,7): error TS2578: Unused '@ts-expect-error' directive.
Error: packages/babel-generator/src/generators/flow.ts(550,3): error TS2578: Unused '@ts-expect-error' directive.
Error: packages/babel-generator/src/generators/flow.ts(555,5): error TS2578: Unused '@ts-expect-error' directive.
```

File: packages/babel-types/package.json (hunk 1 above)
Comment (jedwards1211) [reply packages/babel-types/package.json]: argh, didn't realize tests run on node 8 though, i'll do it callback style

**Comment by nicolo-ribaudo** [comment]
Whops thanks, I'll fix them on `main`

**Comment by nicolo-ribaudo** [comment]
No, but we still set it for consistency with the other export and import declarations.

**Comment by jedwards1211** [comment]
@nicolo-ribaudo okay, well it sets `exportKind: "value"` when parsing TS, but not when parsing Flow, so I think that's a bug with parsing Flow then

**Comment by nicolo-ribaudo** [comment]
@jedwards1211 The linting failures seem to be caused but this PR, but it's a good thing since we were just telling TS "we know that our code doesn't respect the declared type, so there is an error here".

You can just delete those three `@ts-expect-error` comments.

**Comment by jedwards1211** [comment]
@nicolo-ribaudo I believe you but it seems insane...do you think that one change to `yarn.lock` would somehow be affecting `eslint`'s globbing?  I can't think of any other possible explanation.  Usually when I've encountered inexplicable behavior like this it's a symptom of a deeper problem

**Comment by jedwards1211** [comment]
Oh now I get it, someone put those `@ts-expect-error`s there to suppress valid errors due to the missing fields.
Hopefully the fields weren't intentionally missing for some reason?

**Comment by nicolo-ribaudo** [comment]
It's just that we use Flow and TS nodes in the plugins way less than JS nodes, so we never noticed it. Thanks for this PR!

**Review by nicolo-ribaudo — APPROVED** [review]

**Review by JLHwung — APPROVED** [review]

**Pull request merged**